src/Analyzers/VisualBasic/CodeFixes/ConvertTypeOfToNameOf/VisualBasicConvertTypeOfToNameOfCodeFixProvider.vb

```diff
@@ -0,0 +1,35 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Composition
+Imports System.Diagnostics.CodeAnalysis
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.CodeFixes
+Imports Microsoft.CodeAnalysis.ConvertTypeOfToNameOf
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.ConvertTypeOfToNameOf
+
+    <ExportCodeFixProvider(LanguageNames.VisualBasic, Name:=PredefinedCodeFixProviderNames.ConvertTypeOfToNameOf), [Shared]>
+    <ExtensionOrder(After:=PredefinedCodeFixProviderNames.ConvertTypeOfToNameOf)>
+    Friend Class VisualBasicConvertTypeOfToNameOfCodeFixProvider
+        Inherits AbstractConvertTypeOfToNameOfCodeFixProvider
+
+        <ImportingConstructor>
+        <SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification:="Used in test code: https://github.com/dotnet/roslyn/issues/42814")>
+        Public Sub New()
+        End Sub
+        Protected Overrides Function GetSymbolType(node As SyntaxNode, semanticModel As SemanticModel) As ITypeSymbol
+            If node.GetType() Is GetType(MemberAccessExpressionSyntax) Then
```

Comment:
1. `if typeof node is memberAccessExpressoinSytnax`
2. you shouldn't have to actually check.  we only offer the fix if teh anlayzer ran, and the analyzer already checked teh shape.  so it should be safe to just DirectCast.

Reply:
+1 to just calling `DirectCast(node, MemberAccessExpressionSyntax).Expression`

---
In reply to: [456093157](https://github.com/dotnet/roslyn/pull/45806#discussion_r456093157) [](ancestors = 456093157)